files-csharp/CountFilesInaFolderUsingCSharp/CountFilesInaFolderBenchmarks/FileCountingBenchmark.cs

```diff
@@ -0,0 +1,76 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Order;
+using CountFilesInaFolderLibrary.AdvancedApproach.UsingLINQWithDirectoryEnumerateFiles;
+using CountFilesInaFolderLibrary.AdvancedApproach.UsingWinAPI;
+using CountFilesInaFolderLibrary.BasicApproach;
+
+namespace CountFilesInaFolderBenchmarks;
+
+[MemoryDiagnoser]
+[Orderer(SummaryOrderPolicy.FastestToSlowest)]
+public class FileCountingBenchmark
+{
+    private string _directoryPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+
+    [Params(1000, 5000, 10000)]
+    public int FileCount;
+
+    public DirectoryInfo CreateSimulatedDirectory(int fileCount)
+    {
+        var directory = Directory.CreateDirectory(_directoryPath);
+
+        for (int i = 0; i < fileCount; i++)
+        {
+            var filePath = Path.Combine(directory.FullName, $"File{i}.txt");
+            File.WriteAllText(filePath, "Test content");
```

Comment:
This will execute faster and be less disk intensive if we just create empty (0 byte) files. We can do that like this:
`File.Create(filePath).Dispose();`